src/dsp/rx_fft.h

```diff
@@ -94,6 +94,7 @@ class rx_fft_c : public gr::sync_block
 
 private:
     unsigned int d_fftsize;   /*! Current FFT size. */
+    unsigned int n_fftsize;   /*! New FFT size. */
```

Comment:
Gqrx follows [GNU Radio's coding conventions](https://github.com/gnuradio/greps/blob/master/grep-0001-coding-guidelines.md#classes), including:

> All class data members shall begin with `d_foo`
>
> The big benefit is when you're staring at a block of code it's obvious which of the things being assigned to persist outside of the block.

Reply:
Perhaps `d_new_fftsize` would be a better name.

But could you also explain why this new member variable is needed?

Reply:
Also, why is it needed for `rx_fft_c` but not `rx_fft_f`?

I'd like to keep these classes as similar as possible, but the changes here are making them diverge.

Reply:
> why this new member variable is needed?

I want to unlock the mutex as soon, as possible to not block the GNU Radio thread for the time of FFT calculation. This solves audio dropouts and reduces tuning lag when the FFT size is set to a large number (512k..1M) or when the CPU is not capable of calculating the FFT in reasonable time (Atom N270 for example).
The variable is introduced to avoid race between `get_fft_data` and `set_fft_size` when `get_fft_data` is executing in different thread. This commit was picked from my different branch, where I tried to offload fft calculation to a dedicated thread. This worked very well, but it turns out that the rendering of waterfall/spectrum is next heavy task too be offloaded to the dedicated thread, so no PR yet.
I think, this change is unrelated and should be moved to that separate branch.
And mutex unlocking point change should be moved to a separate commit/PR as it may need more testing (see my first comment). Maybe we should apply the window under locked mutex and unlock it before actual FFT calculation...

I'll remove this stuff and push a clean version. Wait some minutes, please.